### PR TITLE
Remove `--template=skeleton` and add a placeholder error message instead.

### LINKED ISSUE #160673: Remove 'skeleton' template
I was planning on updating 'skeleton' to reflect the website's new [app architecture recommendations](https://docs.flutter.dev/app-architecture), but over the past 30 days about 5000 out of ~1m usages of `flutter create -t` used 'skeleton'.  Should we deprecate it instead?

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/160673.

Does the following:

- Renames `FlutterProjectType` to `FlutterTemplateType`; did some enhanced enum cleanups while at it
- Creates a hierarchy of `RemovedFlutterTemplateType` from `ParsedFlutterTemplateType`
- Removes the `skeleton` directory
- Merges `app_shared` back into `app` (no longer required now that `skeleton` is removed)

Final cleanups are tracked in https://github.com/flutter/flutter/issues/160692.

(Added @zanderso just to spot check this is what he meant by https://github.com/flutter/flutter/issues/160673#issuecomment-2557742347)

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
Looks good from my end but I would let @andrewkolos take a look when he is available.

**Comment by matanlurey** [comment]
Some hardcoded-test assumptions still need to be updated. I'll fix those up Monday.

**Review by andrewkolos — APPROVED** [review]
Implementation LGTM. Feel free to re-request if tests end up needing significant re-writes

File: packages/flutter_tools/lib/src/flutter_project_metadata.dart [hunk 1]
```diff
@@ -12,73 +13,122 @@ import 'project.dart';
...
+///
+/// The CLI can give a hint to a developer that the [cliName] _did_ use to exist,
+/// but does no longer, and provides [helpText] for other resources to use instead.
+enum RemovedFlutterTemplateType implements ParsedFlutterTemplateType {
+  skeleton(
+    helpText:
+        'Formerly generated a list view / detail view Flutter application that '
+        'followed some community best practices. For up to date resources, see '
+        'https://flutter.github.io/samples, https://docs.flutter.dev/codelabs, '
+        'and external resources such as https://flutter-builder.app/.',
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/flutter_project_metadata.dart:57]: nit: external -> "community" or "third-party".

File: packages/flutter_tools/lib/src/flutter_project_metadata.dart [hunk 2]
```diff
@@ -12,73 +13,122 @@ import 'project.dart';
...
-  /// This is a Dart package project with external builds for native components.
-  packageFfi,
+/// A [ParsedFlutterTemplateType] that is no longer operable.
+///
+/// The CLI can give a hint to a developer that the [cliName] _did_ use to exist,
+/// but does no longer, and provides [helpText] for other resources to use instead.
+enum RemovedFlutterTemplateType implements ParsedFlutterTemplateType {
+  skeleton(
+    helpText:
+        'Formerly generated a list view / detail view Flutter application that '
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/flutter_project_metadata.dart:54]: @ericwindmill sgty?

**Review by zanderso — COMMENTED** [review]
Yes, this is what I was thinking.

**Pull request merged**